includes/sanitizers/class-amp-block-sanitizer.php

```diff
@@ -41,27 +41,30 @@ public function sanitize() {
 			$node = $nodes->item( $i );
 
 			// We are only looking for <figure> elements which have wp-block-embed as class.
-			$class = (string) $node->getAttribute( 'class' );
-			if ( false === strpos( $class, 'wp-block-embed' ) ) {
+			$classes = preg_split( '/\s+/', trim( $node->getAttribute( 'class' ) ) );
+
+			if ( ! in_array( 'wp-block-embed', $classes, true ) ) {
 				continue;
 			}
 
-			// Remove classes like wp-embed-aspect-16-9 since responsive layout is handled by AMP's layout system.
 			$responsive_width  = null;
 			$responsive_height = null;
-			$node->setAttribute(
-				'class',
-				preg_replace_callback(
-					'/(?<=^|\s)wp-embed-aspect-(?P<width>\d+)-(?P<height>\d+)(?=\s|$)/',
-					function ( $matches ) use ( &$responsive_width, &$responsive_height ) {
+
+			// Remove classes related to aspect ratios as the embed's responsiveness will be handled by AMP's layout system.
+			$classes = array_filter(
+				$classes,
+				static function ( $class ) use ( &$responsive_width, &$responsive_height ) {
+					if ( preg_match( '/^wp-embed-aspect-(?P<width>\d+)-(?P<height>\d+)$/', $class, $matches ) ) {
 						$responsive_width  = $matches['width'];
 						$responsive_height = $matches['height'];
-						return '';
-					},
-					$class
-				)
+						return false;
+					}
+					return 'wp-has-aspect-ratio' !== $class;
+				}
 			);
 
+			$node->setAttribute( 'class', implode( ' ', array_filter( $classes ) ) );
```

Comment:
Minor thing, but the `array_filter()` here can now be removed.

```suggestion
			$node->setAttribute( 'class', implode( ' ', $classes ) );
```